.github/workflows/tests.yml

```diff
@@ -0,0 +1,103 @@
+name: Ruby
+on:
+  push:
+    branches: [ main, develop ]
+  pull_request:
+    branches: [ main, develop ]
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:12
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: ci_test
+        ports:
+        - 5432:5432
+        # needed because the postgres container does not provide a healthcheck
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+    steps:
+    - name: Increase the amount of inotify watchers
+      run: |
+        sudo sh -c "echo fs.inotify.max_user_watches=524288 >> /etc/sysctl.conf"
+        sudo sysctl -p
+    - name: Checkout repository and submodules
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1.90.0
+      with:
+        ruby-version: 3.1
+        bundler-cache: true
+    - uses: actions/cache@v2
+      with:
+        path: vendor/bundle
+        key: ${{ runner.os }}-gems--${{ hashFiles('**/Gemfile.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-gems--
+    - name: Install gems
+      run: |
+        gem install bundler
+        bundle config path vendor/bundle
+        bundle install --jobs 4 --retry 3
+    - name: Get yarn cache directory path
+      id: yarn-cache-dir-path
+      run: echo "::set-output name=dir::$(yarn cache dir)"
+    - uses: actions/cache@v2
+      id: yarn-cache
+      with:
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+    - name: Install packages
+      if: steps.yarn-cache.outputs.cache-hit != 'true'
+      run: yarn install --check-files
+    - name: Setup Database
+      env:
+        RAILS_ENV: test
+        POSTGRES_HOST: 127.0.0.1
+        POSTGRES_USER: postgres
+        POSTGRES_PASSWORD: postgres
+        POSTGRES_PORT: ${{ job.services.postgres.ports[5432] }}
+      run: |
+        cp config/database.ci.yml config/database.yml
+        bundle exec rails db:schema:load
+    - name: Tests
+      env:
+        RAILS_ENV: test
+        POSTGRES_HOST: 127.0.0.1
+        POSTGRES_USER: postgres
+        POSTGRES_PASSWORD: postgres
+        POSTGRES_PORT: ${{ job.services.postgres.ports[5432] }}
+      run: |
+        bundle exec rails assets:precompile
```

Comment:
I'd say assets pre-compilation should probably be another step
I was expecting to see the test report but instead first thing I see was the pre-compilation output
![Uploading Screen Shot 2022-01-21 at 17.26.53.png…]()